core/src/commonMain/kotlin/entity/User.kt

```diff
@@ -48,8 +54,31 @@ public open class User(
 
     /**
      * The 4-digit code at the end of the user's discord tag.
+     *
+     * `"0"` indicates that this user has been migrated to the new username system, see the
+     * [Discord Developer Documentation](https://discord.com/developers/docs/change-log#unique-usernames-on-discord) for
+     * details.
      */
-    public val discriminator: String get() = data.discriminator
+    // "0" when data.discriminator is missing: if the field is missing, all users were migrated,
+    // see https://discord.com/developers/docs/change-log#identifying-migrated-users:
+    // "After migration of all users is complete, the discriminator field may be removed."
+    @Suppress("DEPRECATION", "DeprecatedCallableAddReplaceWith")
+    @Deprecated(
+        "Discord's username system is changing and discriminators are being removed, see " +
+            "https://discord.com/developers/docs/change-log#unique-usernames-on-discord for details.",
+        level = WARNING,
+    )
+    public val discriminator: String get() = data.discriminator.value ?: "0"
```

Comment:
should we even deprecate this? bots will continue to have discriminators...

Reply:
It's better to push toward the new system so it's good to have that. [DeprecatedSinceKord](https://github.com/kordlib/kord/blob/0.9.x/common/src/commonMain/kotlin/annotation/Annotations.kt#LL82C25-L82C44) could be used.